schema/schema_test.go

```diff
@@ -0,0 +1,175 @@
+package schema_test
+
+import (
+	"bufio"
+	"bytes"
+	"io/ioutil"
+	"testing"
+
+	"github.com/hpcloud/termui/schema"
+
+	term "github.com/cloudfoundry/cli/cf/terminal"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// TestSchema tests schema package
+func TestSchema(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	RunSpecs(t, "Schema")
+}
+
+var _ = Describe("Schema", func() {
+	Describe("Test if schema asks correctly for a string", func() {
+		Context("ask for string", func() {
+			It("should be Insert string value for /stringtest [required]", func() {
+
+				str := `{
+   "type":"object",
+   "properties":{
+      "stringtest":{
+         "type":"string"
+      }
+   },
+   "required":[
+      "stringtest"
+   ]
+}`
+
+				in := bytes.Buffer{}
+				out := bytes.Buffer{}
+				writer := bufio.NewWriter(&out)
+
+				ui := term.NewUI(
+					&in,
+					writer,
+					term.NewTeePrinter(writer),
+					nil,
+				)
+
+				schema.NewSchemaParser(ui).ParseSchema(str)
+				writer.Flush()
+
+				result, err := ioutil.ReadAll(&out)
+
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result).To(ContainSubstring("Insert string value for /stringtest [required]"))
```

Comment:
You don't need the `ioutil.ReadAll()` anymore now that you can pass in a `io.Writer`; this is just `Expect(out.String()).To(Contain…)`